src/main/java/hw21/ProductsHandler.java

```diff
@@ -0,0 +1,55 @@
+package hw21;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+public class ProductsHandler {
+
+    public static List<Product> findExpensiveBooks(List<Product> products) {
+        Objects.requireNonNull(products);
+        return products.stream()
+                .filter(product -> product.isBook() && product.getPrice() > 250)
+                .toList();
+    }
+
+    public static List<Product> getBookDiscount(List<Product> products) {
+        Objects.requireNonNull(products);
+        return products.stream()
+                .filter(product -> product.isBook() && product.isDiscount())
+                .peek(product -> product.setPrice(product.getPrice() * 0.9))
```

Comment:
Please replace `peek` with `map` as `peek` is only needed for debugging and testing purposes. If your code won't compile after the change, there is a solution, please find it.